### ISSUE TITLE
CRITICAL: gradient clipping uses ||g||⁴ instead of ||g|| — root cause of 2.7x convergence gap

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#311]
### Bug 1: Per-block norm computation (line 121)

```rust
// squared_sum_collect returns sum(x²) = ||g||²
if let Ok(norm) = squared_sum_collect(p) {
    total_sq += f64::from(norm) * f64::from(norm);  // BUG: squares ||g||² → ||g||⁴
}
```

`squared_sum_collect` already returns the **squared** L2 norm. The code then
squares it AGAIN, computing `||g||⁴` instead of `||g||²`.

Fix: `total_sq += f64::from(norm);` (remove the extra `* f64::from(norm)`)

### Bug 2: LM head norm computation (line 1386-1390)

```rust
let lm_norm = squared_sum_cuda(...);  // Returns ||g||²
let clip_scale = if lm_norm > max_norm {
    max_norm / lm_norm  // BUG: compares ||g||² with max_norm, divides by ||g||²
} else { 1.0 };
```

Should be: `let lm_norm = squared_sum_cuda(...).sqrt();` or compare with `max_norm²`.

### Impact

When `||g|| = 10` and `max_norm = 1.0`:
- **Correct**: `clip = 1.0 / 10 = 0.1` → gradient scaled to norm 1.0
- **Bug 1**: `clip = 1.0 / sqrt(||g||⁴) = 1.0 / 100 = 0.01` → 10x too aggressive
- **Bug 2**: `clip = 1.0 / 100 = 0.01` → 10x too aggressive

During early training (gnorm > 1.0), gradients are clipped **much more
aggressively** than intended. This explains the 2.7x convergence gap:
- PyTorch clips to max_norm=1.0 correctly
- Entrenar clips to max_norm/||g||² — effectively max_norm/||g|| too much

### Evidence

v18 step-for-step comparison:
| Step | PyTorch | Entrenar | Ratio |
|------|---------|----------|-------|
| 0 | 10.65 | 10.62 | 1.0x |
| 100 | 7.87 | 9.93 | 1.3x |
| 500 | 4.15 | 8.02 | 1.9x |
| 3K | 2.33 | 6.40 | 2.7x |

The ratio grows because as loss drops, gradients get smaller, clipping
becomes less aggressive, and the bug matters less — but by then the
model is stuck in a suboptimal basin.

### References

- [PyTorch clip_grad_norm_](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html)
- entrenar#309 (parent issue)
- C-INIT-001 (init fix, done), C-CAUSAL-001 (mask fix, done)
